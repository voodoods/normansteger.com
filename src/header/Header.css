.Header {
  padding: 15px 0;
}

.Header h1,
.Header h2 {
  display: inline;
  font-size: 1.6em;
}

.Header #logo {
  width: 50px;
  float:left;
}

.Header .title {
  float: left;
  margin-top:10px;
}

.Header .title h1 { 
    color: #333;
    margin: 10px;
}
.Header .title h2 { 
    color: #ddd; 
}