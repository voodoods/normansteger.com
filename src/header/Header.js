import React, { Component } from 'react'
import SocialContacts from '../social-contacts/SocialContacts'
import logo from './ns-logo2016.svg'
import StyledHeader from './Header.styled'

class Header extends Component {
  render() {
    return (
      <StyledHeader className="Header">
        <div className="title">
          <a href="/">
            <img id="logo" src={logo} alt="Logo - Norman Steger, Software Developer" title="Home" />
          </a>
          <h1>Norman Steger</h1>&nbsp;
          <h2>UI / UX Designer &middot; Software Developer &middot; Frontend Engineer</h2>
        </div>
        <SocialContacts />
      </StyledHeader>
    )
  }
}

export default Header
