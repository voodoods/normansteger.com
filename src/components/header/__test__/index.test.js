import React from 'react'
import { shallow } from 'enzyme'
import Header from '../'

describe('<Header /> component index', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<Header />)
    expect(wrapper).toMatchSnapshot()
  })
})
