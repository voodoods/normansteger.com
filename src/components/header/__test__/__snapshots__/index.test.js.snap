// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> component index renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <div
          className="title pure-u-lg-19-24 pure-u-md-16-24 pure-u-sm-1"
        >
          <a
            href="/"
          >
            <img
              alt="Logo - Norman Steger, Software Developer"
              id="logo"
              src="ns-logo2016.svg"
              title="Home"
            />
          </a>
          <h1
            className="pure-u-lg-5-24 pure-u-md-5-24 pure-u-sm-1"
          >
            Norman Steger
          </h1>
           
          <h2
            className="pue-u-lg-19-24 pure-u-md-19-24 pure-u-sm-1"
          >
            UI / UX Designer · Software Developer · Frontend Engineer
          </h2>
        </div>,
        <div
          className="socials pure-u-lg-5-24 pure-u-md-8-24 pure-u-sm-1"
        >
          <SocialContacts />
        </div>,
      ],
      "className": "Header pure-u-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <a
              href="/"
            >
              <img
                alt="Logo - Norman Steger, Software Developer"
                id="logo"
                src="ns-logo2016.svg"
                title="Home"
              />
            </a>,
            <h1
              className="pure-u-lg-5-24 pure-u-md-5-24 pure-u-sm-1"
            >
              Norman Steger
            </h1>,
            " ",
            <h2
              className="pue-u-lg-19-24 pure-u-md-19-24 pure-u-sm-1"
            >
              UI / UX Designer · Software Developer · Frontend Engineer
            </h2>,
          ],
          "className": "title pure-u-lg-19-24 pure-u-md-16-24 pure-u-sm-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                alt="Logo - Norman Steger, Software Developer"
                id="logo"
                src="ns-logo2016.svg"
                title="Home"
              />,
              "href": "/",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Logo - Norman Steger, Software Developer",
                "id": "logo",
                "src": "ns-logo2016.svg",
                "title": "Home",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "a",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Norman Steger",
              "className": "pure-u-lg-5-24 pure-u-md-5-24 pure-u-sm-1",
            },
            "ref": null,
            "rendered": "Norman Steger",
            "type": "h1",
          },
          " ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "UI / UX Designer · Software Developer · Frontend Engineer",
              "className": "pue-u-lg-19-24 pure-u-md-19-24 pure-u-sm-1",
            },
            "ref": null,
            "rendered": "UI / UX Designer · Software Developer · Frontend Engineer",
            "type": "h2",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <SocialContacts />,
          "className": "socials pure-u-lg-5-24 pure-u-md-8-24 pure-u-sm-1",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  position: fixed;
  width: 100%;
  background: #333;

  @media (max-width: 768px) {
    position: static;

    a {
      display: inline-block;
    }
  }

  .title {
    display: inline-flex;
    align-items: center;
    padding: 15px 3em;
    height: 10%;
    h1,
    h2 {
      display: inline;
      font-size: 1em;
      line-height: 1.5em;
      padding: 0 0 0 15px;
    }

    h1 {
      color: #fff;
    }
    h2 {
      color: #aaa;
    }

    #logo {
      width: 50px;
    }
  }

  .socials {
    display: inline-flex;
    align-items: center;
    justify-content: flex-end;
    padding: 15px 3em;
    height: 10%;
  }

  @media (max-width: 768px) {
    .title.pure-u-md-1,
    .title.pure-u-sm-1 {
      display: block;
      height: auto;
      padding: 1em;
      text-align: center;

      h1,
      h2 {
        display: block;
        padding: 0;
        margin: 0;
      }
      h1 {
        font-size: 1.5em;
      }

      #logo {
        width: 75px;
      }
    }

    .SocialContacts {
      float: right;
      padding: 15px 3em;
      height: 10%;

      li {
        display: inline-block;
        margin-right: 5px;

        a {
          color: #c1ccd1;
          transition: 0.2s color ease-in-out;
          &:hover {
            color: white;
          }
        }
      }
    }

    .socials {
      justify-content: center;
    }

    .pure-u-sm-1 {
      width: 100%;
      text-align: center;

      .SocialContacts {
        display: inline-block;
        float: none;
        margin: auto;
        padding-top: 0;
      }
    }
  }
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <div
            className="title pure-u-lg-19-24 pure-u-md-16-24 pure-u-sm-1"
          >
            <a
              href="/"
            >
              <img
                alt="Logo - Norman Steger, Software Developer"
                id="logo"
                src="ns-logo2016.svg"
                title="Home"
              />
            </a>
            <h1
              className="pure-u-lg-5-24 pure-u-md-5-24 pure-u-sm-1"
            >
              Norman Steger
            </h1>
             
            <h2
              className="pue-u-lg-19-24 pure-u-md-19-24 pure-u-sm-1"
            >
              UI / UX Designer · Software Developer · Frontend Engineer
            </h2>
          </div>,
          <div
            className="socials pure-u-lg-5-24 pure-u-md-8-24 pure-u-sm-1"
          >
            <SocialContacts />
          </div>,
        ],
        "className": "Header pure-u-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <a
                href="/"
              >
                <img
                  alt="Logo - Norman Steger, Software Developer"
                  id="logo"
                  src="ns-logo2016.svg"
                  title="Home"
                />
              </a>,
              <h1
                className="pure-u-lg-5-24 pure-u-md-5-24 pure-u-sm-1"
              >
                Norman Steger
              </h1>,
              " ",
              <h2
                className="pue-u-lg-19-24 pure-u-md-19-24 pure-u-sm-1"
              >
                UI / UX Designer · Software Developer · Frontend Engineer
              </h2>,
            ],
            "className": "title pure-u-lg-19-24 pure-u-md-16-24 pure-u-sm-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  alt="Logo - Norman Steger, Software Developer"
                  id="logo"
                  src="ns-logo2016.svg"
                  title="Home"
                />,
                "href": "/",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "Logo - Norman Steger, Software Developer",
                  "id": "logo",
                  "src": "ns-logo2016.svg",
                  "title": "Home",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "a",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Norman Steger",
                "className": "pure-u-lg-5-24 pure-u-md-5-24 pure-u-sm-1",
              },
              "ref": null,
              "rendered": "Norman Steger",
              "type": "h1",
            },
            " ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "UI / UX Designer · Software Developer · Frontend Engineer",
                "className": "pue-u-lg-19-24 pure-u-md-19-24 pure-u-sm-1",
              },
              "ref": null,
              "rendered": "UI / UX Designer · Software Developer · Frontend Engineer",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <SocialContacts />,
            "className": "socials pure-u-lg-5-24 pure-u-md-8-24 pure-u-sm-1",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  position: fixed;
  width: 100%;
  background: #333;

  @media (max-width: 768px) {
    position: static;

    a {
      display: inline-block;
    }
  }

  .title {
    display: inline-flex;
    align-items: center;
    padding: 15px 3em;
    height: 10%;
    h1,
    h2 {
      display: inline;
      font-size: 1em;
      line-height: 1.5em;
      padding: 0 0 0 15px;
    }

    h1 {
      color: #fff;
    }
    h2 {
      color: #aaa;
    }

    #logo {
      width: 50px;
    }
  }

  .socials {
    display: inline-flex;
    align-items: center;
    justify-content: flex-end;
    padding: 15px 3em;
    height: 10%;
  }

  @media (max-width: 768px) {
    .title.pure-u-md-1,
    .title.pure-u-sm-1 {
      display: block;
      height: auto;
      padding: 1em;
      text-align: center;

      h1,
      h2 {
        display: block;
        padding: 0;
        margin: 0;
      }
      h1 {
        font-size: 1.5em;
      }

      #logo {
        width: 75px;
      }
    }

    .SocialContacts {
      float: right;
      padding: 15px 3em;
      height: 10%;

      li {
        display: inline-block;
        margin-right: 5px;

        a {
          color: #c1ccd1;
          transition: 0.2s color ease-in-out;
          &:hover {
            color: white;
          }
        }
      }
    }

    .socials {
      justify-content: center;
    }

    .pure-u-sm-1 {
      width: 100%;
      text-align: center;

      .SocialContacts {
        display: inline-block;
        float: none;
        margin: auto;
        padding-top: 0;
      }
    }
  }
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
